### PR TITLE
Bmb/required question update fix

### PR DESCRIPTION
**Pull request by biyeun**
add required styling in preview
fix auto update in preview

@benrudolph / @orangejenny cc @mkangia code buddy

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 7 errors:

* corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js, line 91 - 'Formplayer' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js, line 100 - 'Formplayer' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js, line 101 - Missing trailing comma. (comma-dangle)
* corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js, line 102 - Missing trailing comma. (comma-dangle)
* corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js, line 529 - Expected indentation of 8 space characters but found 12. (indent)
* corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js, line 532 - Expected indentation of 8 space characters but found 12. (indent)
* corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js, line 535 - Expected indentation of 8 space characters but found 12. (indent)

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js [hunk 1]
```diff
@@ -519,19 +520,19 @@ function getEntry(question) {
         case Formplayer.Const.STRING:
             rawStyle = question.style ? ko.utils.unwrapObservable(question.style.raw) === 'numeric' : false;
             if (rawStyle) {
-                entry = new PhoneEntry(question, { enableRateLimit: true });
+                entry = new PhoneEntry(question, { enableAutoUpdate: true });
```
Comment (mkangia) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js]: Why was this needed?

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js [hunk 2]
```diff
@@ -87,20 +87,21 @@ EntrySingleAnswer = function(question, options) {
...
-    self.rawAnswer = ko.observable(question.answer() || Formplayer.Const.NO_ANSWER)
-        .extend(extensions);
+    self.rawAnswer = ko.observable(question.answer() || Formplayer.Const.NO_ANSWER);
 
     self.rawAnswer.subscribe(self.onPreProcess.bind(self));
+
+    var displayOptions = _getDisplayOptions(question);
+    if (options.enableAutoUpdate && ko.utils.unwrapObservable(displayOptions.phoneMode)) {
+        self.valueUpdate = 'keyup';
+        self.answer.extend({
```
Comment (biyeun) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js]: no it should definitely be `answer`. that was the cause of the bug

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js (hunk 1 above)
Comment (biyeun) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js]: clarity

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js [hunk 3]
```diff
@@ -87,20 +87,21 @@ EntrySingleAnswer = function(question, options) {
...
-        self.valueUpdate = 'keyup';
-    }
-    self.rawAnswer = ko.observable(question.answer() || Formplayer.Const.NO_ANSWER)
-        .extend(extensions);
+    self.rawAnswer = ko.observable(question.answer() || Formplayer.Const.NO_ANSWER);
 
     self.rawAnswer.subscribe(self.onPreProcess.bind(self));
+
+    var displayOptions = _getDisplayOptions(question);
+    if (options.enableAutoUpdate && ko.utils.unwrapObservable(displayOptions.phoneMode)) {
```
Comment (benrudolph) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js]: one thing that make make this cleaner, is to only have `enabledAutoUpdate` be `true` if `phoneMode` is true when we pass in the option to the control

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js (hunk 2 above)
Comment (mkangia) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js]: Oh ok.

**Comment by biyeun** [comment]
@benrudolph updated

**Comment by dimagimon** [comment]
There are 7 errors:

* corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js, line 91 - 'Formplayer' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js, line 99 - 'Formplayer' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js, line 100 - Missing trailing comma. (comma-dangle)
* corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js, line 101 - Missing trailing comma. (comma-dangle)
* corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js, line 531 - Expected indentation of 8 space characters but found 12. (indent)
* corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js, line 534 - Expected indentation of 8 space characters but found 12. (indent)
* corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js, line 537 - Expected indentation of 8 space characters but found 12. (indent)

**Review by benrudolph — APPROVED** [review]
👍 on ✔️  (would be nice to clean up a couple of the dimagimon if possible, though realize some weren't you)

**Comment by biyeun** [comment]
@benrudolph ✅
